consensus/ethash/ethash.go

```diff
@@ -250,7 +250,9 @@ func (lru *lru) get(epoch uint64, epochLength uint64, ecip1099FBlock *uint64) (i
 	}
 
 	// Update the 'future item' if epoch is larger than previously seen.
-	if epoch < maxEpoch-1 && lru.future < nextEpoch {
+	// Last conditional clause ('lru.future > nextEpoch') handles the ECIP1099 case where
+	// the next epoch is expected to be LESSER THAN that of the previous state's future epoch number.
+	if epoch < maxEpoch-1 && (lru.future < nextEpoch || lru.future > nextEpoch) {
```

Comment:
I wonder if this, will be easier to read `if epoch < maxEpoch-1 && lru.future != nextEpoch {` or harder to understand the intention of the code.

Reply:
Ah, nice catch! I think your way is readable and understandable too. Will patch.

Reply:
Resolved with https://github.com/etclabscore/core-geth/pull/499/commits/bac80dae1ea98d4acae701077c9f09d78dd406d0